common/include/pcl/common/distances.h

```diff
@@ -197,6 +197,125 @@ namespace pcl
+  };
+
+  /**
+    * \brief Get the distance between a point and line
+    * \param[in] p1 Origin point of the line
+    * \param[in] p2 Terminating point of the line
+    * \param[in] p The point for calulating the distance to the line
+    * \return DistPoint2LineResults
+    */
+  PCL_EXPORTS inline PointToLineSegmentDistanceResults
+  pointToLineSegmentDistance (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2, const Eigen::Vector3f &p)
+  {
+    PointToLineSegmentDistanceResults results;
+
+    results.points[0] = p1;
+    results.points[1] = p2;
+    results.points[2] = p;
+
+    Eigen::Vector3f v = results.points[1] - results.points[0];
+    Eigen::Vector3f w = p - results.points[0];
+
+    double c1 = w.dot (v);
+    double c2 = v.dot (v);
+
+    if (c1 <= 0)
+      results.mu = 0.0;
+    else if (c2 <= c1)
+      results.mu = 1.0;
+    else
+      results.mu = c1 / c2;
+
+    results.p = results.points[0] + results.mu * v;
+    results.d = (p - results.p).norm ();
+
+    return results;
+  }
+
+  struct LineSegmentToLineSegmentDistanceResults
+  {
+    Eigen::Vector3f points[4]; /**< \brief The line segment points. Note: Seg1 = (points[1] - points[0]) and Seg2 = (points[3] - points[2]) */
+    Eigen::Vector3f p[2];      /**< \brief The closest point on each line segment. Note: p[0] is the closet point on Seg1. */
+    float mu[2];               /**< \brief The corresponding line paramentric value resulting in the minimum distance. */
+    double d;                  /**< \brief The distance between both line segments */
+    bool parallel;             /**< \brief Indicate that both lines are parallel. */
+  };
+
+  /**
+    * \brief Distance between two line segments
+    *
+    * Follow the link for more details http://paulbourke.net/geometry/pointlineplane/
+    * \param[in] p1 Origin point of the first line
+    * \param[in] p2 Terminating point of the first line
+    * \param[in] p3 Origin point of the second line
+    * \param[in] p4 Terminating point of the second line
+    * \return DistLine2LineResults
+    */
+  PCL_EXPORTS inline LineSegmentToLineSegmentDistanceResults
+  lineSegmentToLineSegmentDistance (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2, const Eigen::Vector3f &p3, const Eigen::Vector3f &p4)
+  {
+    LineSegmentToLineSegmentDistanceResults result;
+    result.points[0] = p1;
+    result.points[1] = p2;
+    result.points[2] = p3;
+    result.points[3] = p4;
+
+    double d1321 = (p1 - p3).dot ((p2 - p1));
+    double d2121 = (p2 - p1).dot ((p2 - p1));
+    double d4321 = (p4 - p3).dot ((p2 - p1));
+    double d1343 = (p1 - p3).dot ((p4 - p3));
+    double d4343 = (p4 - p3).dot ((p4 - p3));
+
+    double denom = d2121 * d4343 - d4321 * d4321;
+    if (denom < 1.0e-8) // The lines are parallel
+    {
+      PointToLineSegmentDistanceResults temp, dist;
+      result.parallel = true;
+
+      dist = pointToLineSegmentDistance (p1, p2, p3);
+      result.mu[0] = dist.mu;
```

Comment:
You only need to compute the point to line segment once because they're parallel. In this situation, you should set both `mu`s and the corresponding interception points to NaN